internal/graph/check.go

```diff
@@ -572,62 +572,182 @@ func checkAssociatedObjects(ctx context.Context, req *ResolveCheckRequest, objec
 	}, nil
 }
 
-// checkUsersetSlowPath will check userset or public wildcard path.
-// This is the slow path as it requires dispatch on all its children.
-func (c *LocalChecker) checkUsersetSlowPath(ctx context.Context, req *ResolveCheckRequest, iter *storage.ConditionsFilteredTupleKeyIterator) (*ResolveCheckResponse, error) {
-	ctx, span := tracer.Start(ctx, "checkUsersetSlowPath")
-	defer span.End()
-	var handlers []CheckHandlerFunc
+type dispatchParams struct {
+	parentReq *ResolveCheckRequest
+	tk        *openfgav1.TupleKey
+}
 
-	typesys, _ := typesystem.TypesystemFromContext(ctx)
+type dispatchMsg struct {
+	err            error
+	shortCircuit   bool
+	dispatchParams *dispatchParams
```

Comment:
Just curious, what drove the decision to use a pointer here?

Reply:
I'm assuming it is because it is an optional value.

Reply:
Yes, this is optional.  If err / shortCircuit, this will be nil